ios/RNGoogleMobileAds/RNGoogleMobileAdsConsentModule.m

```diff
@@ -121,4 +121,17 @@ - (NSString *)getConsentStatusString:(UMPConsentStatus)consentStatus {
 
 RCT_EXPORT_METHOD(reset) { [UMPConsentInformation.sharedInstance reset]; }
 
+RCT_EXPORT_METHOD(getTCString : (RCTPromiseResolveBlock)resolve : (RCTPromiseRejectBlock)reject) {
+  @try {
+    NSString *tcString = [[NSUserDefaults standardUserDefaults] objectForKey:@"IABTCF_TCString"];
```

Comment:
Is this key a constant from the form itself (here and in Java), or some other spot? I wonder if the key could be imported as a symbol so it wasn't just a magic string

More importantly, if we're storing and decoding persistent state on disk, is there any provision for versioning, checking for staleness, updating contents of that string if the form has changed? This may all be a function of the library, I am very new to this domain

Reply:
I sadly couldn't find an export. They refer to using this string inside their documentation as well, I referenced it in the code comments.

We do not interact with the TCString besides reading it from disk. Google and 3rd party vendors use the content of the string to check for available consent and other information. The UMP SDK stores it on the device and also automatically handles outdated values, e.g. if Google itself adds new vendors to their list. The consent modal will again be shown to the user.

The IAB and the Consent Framework are slow moving industry standards, generally speaking, and the key should stay relevant for quite a while.

Reply:
Okay, that makes sense - what I understood was "the libraries handle this", and the format does seem forward-compatible, sort of JSON-y